tests/functional/services/gcs/test_get_collection_list.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from globus_sdk import GCSAPIError
+from globus_sdk._testing import get_last_request, load_response
+
+
+def test_get_collection_list(client):
+    meta = load_response(client.get_collection_list).metadata
+    res = client.get_collection_list()
+
+    assert len(list(res)) == len(meta["collection_ids"])
+
+    # sanity check some fields
+    assert res["DATA_TYPE"] == "result#1.0.0"
+    for index, item in enumerate(res):
+        assert item["DATA_TYPE"] == "collection#1.0.0"
+        assert item["id"] == meta["collection_ids"][index]
+        assert item["storage_gateway_id"] == meta["gateway_ids"][index]
+        assert item["display_name"] == meta["display_names"][index]
+
+
+def test_get_collection_list_include_param(client):
```

Comment:
It looks like this could be broken into four tests, two of which are parametrized:

1. Test without an `include` parameter
2. Test the `include` parameter
3. Test the `mapped_collection_id`
4. Test the `filter` parameter

Bringing the three `include` parameters and expected values to the top would help highlight what's actually getting checked:

```python
@pytest.mark.parametrize(
    "include, expected",
    (
        ("foo", "foo"),
        ("foo,bar", "foo,bar"),
        (("foo", "bar"), "foo,bar"),
    )
)
def test_get_collection_list_include_param(client, include, expected):
    load_response(client.get_collection_list)
    client.get_collection_list(include=include)
    req = get_last_request()
    assert "include" in req.params
    assert req.params["include"] == expected

Reply:
I copied your sample and modified it very minimally for that case, and fleshed out the other two tests.

The only significant difference between what you suggested and what I did was that I didn't include a separate test for `"include" not in req.params`. I instead made that a case in which `include_param, expected = None, None`. I know it puts conditional logic into the test, but it feels to me like the "same thing" being checked.

(Also, I saw you approved, but I just pushed this, so you may want to voice an opinion at this point. 🙂 )

Reply:
This looks good to me!